### PR TITLE
Add "Play sample sound" to audio setting

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

**Comment by zzmp** [comment]
### QA Plan

- Open the audio settings
- Hit play button next to "Play sample sound"
  - Music should start
  - If you switch devices, music should continue in the correct device
  - If you hit stop, music should stop
  - If you hit start again, music should restart
  - If you close the menu, music should stop
  - If you open the menu and play again, and let it play through, button should change back to a stop when music finishes

NOTE: The music is located where you are when you hit start. If you move too far away, it will grow faint and go away. This is expected behavior.

File: interface/resources/qml/hifi/audio/PlaySampleSound.qml [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+import "../../styles-uit"
+import "../../controls-uit" as HifiControls
+
+RowLayout {
+    property var sound: null;
+    property var sample: null;
+    property bool isPlaying: false;
+    function createSampleSound() {
+	var SOUND = Qt.resolvedUrl("../../../sounds/sample.wav");
```
Comment (zfox23) [inline interface/resources/qml/hifi/audio/PlaySampleSound.qml]: tab

File: interface/src/avatar/MyAvatar.h [hunk 2]
```diff
@@ -56,6 +56,7 @@ class MyAvatar : public Avatar {
      *
      * @namespace MyAvatar
      * @augments Avatar
+     * @property qmlPosition {Vec3} Used as a polyfill for position access by QML
```
Comment (zfox23) [inline interface/src/avatar/MyAvatar.h]: did you actually mean "polyfill" here this time? :)

**Comment by CainFoool** [comment]
Testing.

**Comment by zzmp** [comment]
@CainFoool make sure to wait for the build from 5c0e9e7

**Comment by CainFoool** [comment]
@zzmp Yup - waiting for it :)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/867/HighFidelity-Beta-PR10744-5c0e9e7496f2e4059eb7d5fa750f811662b288b2.exe

**Comment by CainFoool** [comment]
@zzmp: Build failed.

**Comment by zzmp** [comment]
test this please

**Comment by zzmp** [comment]
@CainFoool go ahead and test of off the windows installer. the macOS build took too long, so it was aborted - not because of errors.

**Comment by CainFoool** [comment]
@zzmp: Will do, testing now.

**Comment by CainFoool** [comment]
# Test Failed 

The 'Play' button beside Play Sample Sound, is not a play button. Play
Same for while the music is playing, Playing 

- [x] Music should start
- [x] If you switch devices, music should continue in the correct device
- [x] If you hit stop, music should stop
- [x] If you hit start again, music should restart
- [ ] If you close the menu, music should stop
- [x] If you open the menu and play again, and let it play through, button should change back to a stop when music finishes

The music does not stop when the menu closes, however while the music is playing in the background, when you re-open the menu the music then stops.

###### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/874/HighFidelity-Beta-PR10744-5c0e9e7496f2e4059eb7d5fa750f811662b288b2.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/874/HighFidelity-Beta-PR10744-5c0e9e7496f2e4059eb7d5fa750f811662b288b2.dmg

**Comment by zzmp** [comment]
@CainFoool I am not seeing the same characters as you. I am seeing a play and stop button. Can you clear your AppData, do a fresh install, and see if that resolves?

I *can* reproduce your issue with the audio not stopping when the menu closes. Working on that now.

**Comment by zfox23** [comment]
@zzmp was there an update to the actual hifi glyphs font file that you forgot to commit?

**Comment by CainFoool** [comment]
@zzmp I'll retest now on a different PC to make sure.

**Comment by zzmp** [comment]
@zfox23 1bf16c4318daad5621271663bec4629a7a645a82 should update the fonts.

**Comment by CainFoool** [comment]
# Test Failed 

On a fresh install with AppData Local and Roaming Cleared, the same results with the image are happening.

image
image

##### Intel i5 7600K @ 4Ghz, 16GB DDR4 2133Mhz Corsair Vengeance, Asus Strix GTX 1050Ti 3GB, HTC Vive

**Comment by zzmp** [comment]
@CainFoool please retest

**Comment by CainFoool** [comment]
Will do - testing when ready.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/902/HighFidelity-Beta-PR10744-f55d842fac6c3c885a9d128c87ac9964aab0601b.dmg

**Comment by CainFoool** [comment]
@zzmp: Looks like this build failed for Windows.

**Comment by zzmp** [comment]
all windows build are having issues. will rebuild when that is fixed.

**Comment by zzmp** [comment]
@zfox could you rereview the new changes?

**Comment by zfox23** [comment]
yes!

File: interface/resources/qml/hifi/audio/PlaySampleSound.qml [hunk 3]
```diff
@@ -39,6 +39,7 @@ RowLayout {
 
     Component.onCompleted: createSampleSound();
     Component.onDestruction: stopSound();
+    onVisibleChanged: { if (!visible) stopSound(); }
```
Comment (zfox23) [inline interface/resources/qml/hifi/audio/PlaySampleSound.qml]: Coding standard says use explicit brackets and newlines.

**Comment by zfox23** [comment]
@zzmp LGTM besides the one `if()` statement formatting issue.

**Comment by zzmp** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/917/HighFidelity-Beta-PR10744-278a122291e18fa7e146bd4917a51892a1a81130.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/917/HighFidelity-Beta-PR10744-278a122291e18fa7e146bd4917a51892a1a81130.dmg

**Comment by CainFoool** [comment]
# Test Passed 

- [x] Music should start
- [x] If you switch devices, music should continue in the correct device
- [x] If you hit stop, music should stop
- [x] If you hit start again, music should restart
- [x] If you close the menu, music should stop
- [x] If you open the menu and play again, and let it play through, button should change back to a stop when music finishes

**Pull request merged**